### PR TITLE
Migrate away from OC_App and toward IAppManager.

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

Migrate away from OC_App and toward IAppManager.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by icewind1991 — APPROVED** [review]

**Review by ChristophWurst — APPROVED** [review]
To avoid some kaboom we might want to merge this after the 29 branch-off

**Review by skjnldsv — APPROVED** [review]

**Comment by come-nc** [comment]
Rebased and fixed conflicts

File: core/Command/App/GetPath.php [hunk 1]
```diff
@@ -69,7 +80,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 	 */
 	public function completeArgumentValues($argumentName, CompletionContext $context): array {
 		if ($argumentName === 'app') {
-			return \OC_App::getAllApps();
+			return $this->appManager->getInstalledApps();
```
Comment (nickvergessen) [inline core/Command/App/GetPath.php:83]: This is not an equal replacement.
The old one was iterating over the apps directories. This here is looping over `enabled` appconfig values.
So the command no longer autocompletes a path for an app that was cloned but not yet ever enabled.

File: core/Command/App/Update.php [hunk 2]
```diff
@@ -88,7 +91,7 @@ protected function execute(InputInterface $input, OutputInterface $output): int
 				return 1;
 			}
 		} elseif ($input->getOption('all') || $input->getOption('showonly')) {
-			$apps = \OC_App::getAllApps();
+			$apps = $this->manager->getInstalledApps();
```
Comment (nickvergessen) [inline core/Command/App/Update.php:94]: Same here:
This is not an equal replacement.
The old one was iterating over the apps directories. This here is looping over `enabled` appconfig values.
So the command no longer autocompletes a path for an app that was cloned but not yet ever enabled.

But I guess it's somewhat okay in this case.

Comment (nickvergessen) [reply core/Command/App/Update.php:94]: > Does the command work on a disabled app?

It's updating several apps on our servers we don't use at the moment.

**Comment by come-nc** [comment]
@nickvergessen Reverted both calls. We’ll need later to add a replacement for getAllApps.

But the `maintenance:install` command segfaults since today rebase. I can reproduce locally but have no idea what happens.
I tried reverting the IURLGenerator injection in the appmanager but it does not change anything.

**Comment by joshtrichards** [comment]
Related to open issue: #8505 :)

**Pull request merged**